src/components/UriViewer/UriDownloadButton.js

```diff
@@ -22,6 +22,41 @@ const getMaxDownloadCostNA = bytes => {
   return Utils.formatUSD(downloadPrice)
 }
 
+export const AzureUriDownloadButton = ({ uri, fileName }) => {
+  const [url, setUrl] = useState()
+  const getUrl = () => {
+    if (isAzureUri(uri)) {
+      setUrl(uri)
+    }
+  }
+  useOnMount(() => {
+    getUrl()
+  })
+  return els.cell([
+    url === null ?
+      'Unable to generate download link.' :
+      div({ style: { display: 'flex', justifyContent: 'center' } }, [
+        h(ButtonPrimary, {
+          disabled: !url,
+          onClick: () => {
+            Ajax().Metrics.captureEvent(Events.workspaceDataDownload, {
+              ...extractWorkspaceDetails(workspaceStore.get().workspace),
```

Comment:
UriViewer has a `workspace` prop. To make sure that this download button is using the same workspace as its parent UriViewer, I think it would be better to pass `workspace` down to `AzureUriDownloadButton` instead of getting it from global state.